### ISSUE TITLE
False positive example, can't interact with `isinstance`

### ISSUE THREAD

**Issue by GoToCoding** [python/mypy#9226]
![image](https://user-images.githubusercontent.com/16542586/88675774-f7213380-d0f3-11ea-856c-38492789f4de.png)

code:
```py
from typing import TypeVar, List

StrOrList = TypeVar('StrOrList', str, List[str])

def f(a: StrOrList) -> StrOrList:
    b = a if isinstance(a, list) else [a]
    return b if isinstance(a, list) else b[0]
```

```bash
> mypy ~/tmp/test.py
/home/gotocoding/tmp/test.py:6: error: List item 0 has incompatible type "List[str]"; expected "str"
/home/gotocoding/tmp/test.py:7: error: Incompatible return value type (got "Sequence[str]", expected "List[str]")
```

I got error, but I expect that there should be no errors in this case.

Python 3.7.4, mypy 0.782

**Comment by JukkaL** [comment]
Hmm this may be hitting some mypy bug.

**Comment by AlexWaygood** [comment]
Seems to be a bug relating to the treatment of ternary expressions; a version of the same snippet that doesn't use ternary expressions [works fine](https://mypy-play.net/?mypy=latest&python=3.10&flags=check-untyped-defs&gist=9ea602521b57dead7e41939335908d24):

```python
from typing import TypeVar, List

StrOrList = TypeVar('StrOrList', str, List[str])

def f(a: StrOrList) -> StrOrList:
    if isinstance(a, list):
        b = a
    else:
        b = [a]
    if isinstance(a, list):
        return b
    else:
        return b[0]
```

**Comment by ilevkivskyi** [comment]
This seem to work correctly now.
